components/mdc/Datatable/Datatable.svelte

```diff
@@ -2,24 +2,47 @@
 <script>
 import { MDCDataTable } from '@material/data-table'
 import { createEventDispatcher, onMount } from 'svelte'
-
+/**
+ * @prop {string}
+ * @description used for aria-label
+ */
 export let label = ''
+/**
+ * @prop {any}
+ * @description used to register new Datatable Checkboxes when length changes
+ */
+export let checkboxIsMountedArray = []
 
 const dispatch = createEventDispatcher()
+let dataTable = {}
 
 let element = {}
 
 onMount(() => {
-  const dataTable = new MDCDataTable(element)
+  dataTable = new MDCDataTable(element)
 
   dataTable.listen('MDCDataTable:sorted', (event) => {
     dispatch('sorted', event.detail)
   })
 
+  dataTable.listen('MDCDataTable:selectedAll', () => {
+    dispatch('selectedAll')
+  })
+
+  dataTable.listen('MDCDataTable:unselectedAll', () => {
+    dispatch('unselectedAll')
+  })
+
+  dataTable.listen('MDCDataTable:rowSelectionChanged', (event) => {
+    dispatch('rowSelectionChanged', event.detail)
+  })
+
   // This does not work because of an MDC bug. See https://github.com/material-components/material-components-web/issues/6385
```

Comment:
Looks like this comment may need to be updated. That bug seems to have been fixed, as I'm guessing you found since you uncommented the destroy line.

Reply:
I forgot to delete that. The issue was fixed.